Event Creation

ZachWileman opened the pull request.

Commit:
[UPDATE]

- Added initial event creation page
- Added styling to event page
- Added calendar popup to datetime input fields
- Successfully adds events to the database (while using a temporary
  User grabbed from the database)

Commit:
Merge branch 'master' of github.com:MST-ACM/acm.mst.edu into createEvent

Commit:
[UPDATE] Added more to the Event Creation Page

- After successfully adding an event, you’ll get redirected to the home page now
- Created a new look for the Event Creation page
- Added text area field to “Title” and “Description” field
- Fixed Django block tag on “listEvents.html”

Commit:
[UPDATE] Updated the Create Event page (w/ other misc. things)

[NOTE] When this commit is merged to master, your Vagrant box will need to be destroyed if you have
inserted Events into the database prior to the pulling of this commit. This is because the new field
“flier”(ImageField) added to Event model has no default value and is also required. Django will
ignore it and not add it to the database since it won’t know what to fill the newly added column with
to the events that are currently in the database (if any).

- Added the new field “flier” to the Event model; enables the user to upload images for fliers when
  creating an event.
- Added to the Vagrant and requirements file modules that were needed for the use of the ImageField
  for uploading images.
- Commented out some things in the rest_api app so that ImageField could be used.
- Added “material_blue” style to calendar
- Added “media/“ to .gitignore file
- Removed unnecessary datetime format specifications for the “date_hosted” and “date_expire” widgets
- Added necessary code in “create_event“ view so that image files can be pulled from the form after a
  post request.
- Added function in “events/models.py” file named “get_path_for_fliers” which is used for creating a
  directory for storing media files. Files are stored in the following manner:
  “media/fliers/<event.date_hosted>/<filename>”. Also, the media folder is stored in the same directory
  as all of the apps much like the static folder.
- MEDIA_URL and MEDIA_ROOT were changed in the Django settings.py file to accommodate for the storage
  of media files.
- Added a fix that makes sure the datetime can only be changed by the calendar and also makes the
  Field required.
- Added a check that makes sure the Event expiration date is after the hosting date of the Event. The
  check is done client side with jQuery and pops up an alert if the error occurs. (Was considering
  having the error appear on the page, but it was hard to work around making the page look nice with
  the error since FlexBox likes to auto-adjust columns as needed)

Commit:
Removed a couple of lines that were no longer relevant.

Commit:
Merge branch 'master' of github.com:MST-ACM/acm.mst.edu into createEvent

Commit:
[UPDATE] Updated the front page

- Fixed static references to flier images
- Added box shadows to the flier images. This is done in case the image
  is white and has a white background.

Commit:
[UPDATE]

- Changed the homepage so that it now displays events dynamically.
- The home page displays the 3 closet upcoming events that are greater
  than or equal to the current datetime.

Commit:
[UPDATE/FIX]

- Few small updates to how the events appeared on the front page
- Added a way for grabbing the user's current timezone through use
  of a django app that was added to the installed_apps in the settings.py
  file called "tz_detect"

Commit:
Removing django timezone app for now

Commit:
[UPDATE] Added optional link to Event

- The user can now add an optional link with an event.
- Also added to the homepage to where it now displays a click-able
  link for an event if a link was added to it when the event was created.

Commit:
Merging master into createEvent branch

Commit:
[UPDATE] Added documentation page for Project Apps

- Created basic links to project apps on index page
- Created new file 'apps.txt' file to be used for documenting project
  applications.
- Began working on documentation of Events app. by discussing the 'forms.py'
  file.

Commit:
Create apps.txt

Fixed small typo.

Commit:
[UPDATE] Updated documentation on Events app

- Added toctree to the index page
- Added documentation on the "models.py" and "views.py" file.

Commit:
[UPDATE] Updated documentation and autodoc settings

- Moved information from the apps.txt over to the corresponding
  places in docstrings within the python files.
- Updated Makefile to create a shorthand for creating auto-documented
  rst files along with certain customizations so that only certain autodoc
  tags get added within the auto-generated rst files.
- Made small changes to the conf.py which include:
   - added myself as an author
   - changed theme
   - changed source suffix to .rst instead of .txt
   - added needed django setup as well as added python files path to
     sphinx so that the autodoc generation worked correctly.
- Updated docuentation for various files within the event app.
- Updated .gitignore to ignore the auto-generated rst files since
  they can be created automatically with the new "make rst" command
  that I added.

Commit:
Removed some 'commented out' things from the conf.py file

Commit:
[UPDATE/FIX]

- changed the 'media/fliers' directory name to just 'fliers' due to
  an error occuring regarding the media tab now being used on the home
  page; fliers weren't appearing and was instead being linked to the media
  tab on the home page.
- added 'novalidate' to the create event form for the URLField so that
  when users create events, they don't need to include 'https://' at the
  beginning of the URL. Also, even without the 'novalidate' attribute
  added, the form can still work with incorrect urls
  (ex: https://www.acm.mst.edu).
- Made event creation valid if the current user attempting to create an
  event is a superuser. Also, added a page redirect for users who aren't
  superusers attempting to access the event creation page.
- Added line back to the 'index.html' home page that allowed for events
  to be dynamically added to the front page; must've gotten accidentally
  removed.

Commit:
Removed unnecessary import from views.py

Commit:
[UPDATE] Added new test for events

- Added test for events app that tests the flier field along with
  the get_path_for_flier function.
- Updated Vagrant file to allow for unit tests to work with 'djangouser'

Commit:
[UPDATE/FIX] Updated tests and model for Event

- FIX: Removed the 'novalidate' HTML attribute from the create event HTML
  page because it was causing everthing to not be validated rather than
  just the link field.
- Added TextInput widget to link field so that the webpage no longer
  recognized it as URL so that it no longer tries to validate the URL.
- Added clean method to the Event model so that it can now validate the
  host and expire date via Django rather than using Javascript. Additionally,
  since this is now validated via the clean method, the JS that was previously
  doing this check has been commented-out for now.
- Added to the HTML create event pages so that it now outputs error messages
  pertaining to each of the fields and non-field errors.
- Added css styling for error messages that now appear on the create event page.
- Added to the view file so that if errors occur, the user is redirected back to
  the event page and displays the info that they had entered along with the
  errors pertaining to what they did wrong.
- Added folder named 'test_data' which currently just holds an image used
  for testing puroses.
- Updated "test_get_path_for_flier_function_along_with_image_field" test function
  to accomodate for the new clean method added to the Event model.
- Added tests to "ViewTestCase" which tests the availability of the create
  event page based off of whether or not a user is a superuser.
- Added 'EventFormTestCase' which tests the Event model under different
  circumstances.

Commit:
Fixed small change that occured on accident

Commit:
[UPDATE] Updated the events app

- Inside of the settings, I changed the MEDIA_ROOT and MEDIA_URL to use
  'media_files' as the base folder for storing media files such as fliers.
  Also created a FLIERS_PATH variable for use in different files such as
  in the event model's file for the 'get_path_for_flier' function.
- Changed some of the documentation inside of the forms.py file
- Updated ModelTestCase to use tempdir as well as changes the settings.MEDIA_ROOT
  variable so that files can saved in a temporary directory so that files don't
  interact with where all of the media files that aren't used for tests.
- Updated ViewTestCase so that code coverage is better.
- Updated documentation inside of views file
- Added coverage module to the requirements file

Commit:
Got rid of unnecessary lines of code

kevinschoonover committed:
[UPDATE] Comment fixes

kevinschoonover committed:
[FIX] Fixed failing PUT request for EventDetail

Commit:
[UPDATE] Cleaned up rest api test section

- Cleaned up part of the rest api test section under the EventsTestCase
- Made some small changes to the vagrant file to bring it up to date with
  the develop branch and so to remove the errors occuring that involved the
  outdated vagrant box.

kevinschoonover committed:
[FIX] Solved the clipping of the text in the flatpicker

Commit:
[FIX] Fixed the clipping of the calendar month name on the create-event page

- Changed the font family of the overall body element so that the font
  being used inside the calendar datetime picker would then be changed as
  well; specifically to a font family that doesn't clip out part of the
  month's name.
- The previous fix worked for Firefox only whereas this fix seems to work
  on Firefox, Safari, and Chrome.

Commit:
[FIX] Made a small fix to the Makefile for use of autodoc with sphinx

Commit:
Merge branch 'develop' of github.com:MST-ACM/acm.mst.edu into createEvent

- Also made changes to the Vagrantfile to fix 'apt' and 'pip' error
- Began making changes to adjust for merging of branches.
- Brought back 'core/base.html' file since the create event page was using it.

Commit:
Merge branch 'develop' of github.com:MST-ACM/acm.mst.edu into createEvent

- Also made spelling error fix on the 'upcoming-events' file that
  was a css tag to not be applied correctly.

Commit:
[UPDATE] Changes to the homepage

- Implemented a settings variable which makes it easier to change the
  max number of upcoming events that are being displayed on the homepage.
- Also removed placeholder events from the index.html file as to take
  advantage of the system in place; all that needs to be done is to have
  an event created through the event creation page.

Commit:
Removing unnecessary file

Commit:
[FIX] Removed css styling which was preventing text from showing a shadow when highlighted.

Commit:
[UPDATE] Small documentation change

- Removed general documentation on urls.py file inside of Events app.
  The documentation isn't specific to the event's url file so I have
  removed it from the file but kept it on my computer for when we decide
  to create a general documentation area (if we do).

Commit:
[UPDATE] Updated documentation with sphinx

- Updated Makefile so that sphinx auto generates documentation
  for all apps (events, payments, etc.)
- Also went through various files and updated them so that sphinx
  doesn't give any errors when creating the html files from the
  auto generated rst files.

Commit:
Additional changes that were supposed to go with previous commit

Commit:
Removing redundant url file documentation

Commit:
[UPDATE] Added a README for the docs folder

Commit:
Merge branch 'develop' into createEvent

ClayMav commented:
I just pulled down your branch and it is giving me a ProgrammingError right off the bat. Is this maybe because I don't have any events stored? If so, it should probably handle that.

kevinschoonover commented:
Did you do the migration? @ClayMav

ClayMav commented:
I tried migrate to no avail, and i just tried makemigrations and got this
![nomigrate](https://user-images.githubusercontent.com/16675291/29980620-1fb822e6-8f10-11e7-9318-6b156a1aab3c.png)
Apologies if this is just me not understanding the system.

zdw27f commented:
@ClayMav 

So assuming you have a vagrant box created (which is done via "vagrant up"), you should then run "vagrant reload --provision" which does the migrations for you. I would tell you to do "vagrant destroy" to git rid of the current vagrant box you're using and then create a new one with "vagrant up", but I don't know if you've stored anything in the database in your vagrant box; most likely you haven't since most of your stuff is front end.

So, at the least, I would recommend:
1.) Create a new directory
2.) Clone down my branch into the new directory
3.) Run "vagrant up" inside of the new directory
4.) Followed by "vagrant reload --provision"

This should hopefully at least get rid of whatever error you're getting right now.

There are additional things that are going to have to be done via the Django shell as well as editing the views file inside of the Events app so that you may access the event creation page, but that is because the "create-event" page is restricted to only superusers and you can only login via the main website and doesn't work on your localhost or dev website (assuming you don't have google authentication setup on your localhost).

If you want, I could try to meet with you today to try and explain everything and get it working on your computer but I understand if this is too late of notice. If you do want to meet today, I could do any time after 5pm.

@kevinschoonover I'm tagging Kevin as well in case he wants me to explain what I've done on the create event branch.

ClayMav commented:
FYI:
After a sit-down with Zach, my problem was solved.
Problem was that having changed branches after provisioning the box caused the migrations to stay and ruin things.
After nuking my local files and cloning, it worked.
Also, there was an issue where the mixture of tabs and spaces in the Vagrantfile caused my local vagrant to fail a few times.
I give this request a go

kevinschoonover commented on ACM_General/ACM_General/settings.py:
```diff
@@ -19,10 +19,15 @@
+MEDIA_URL = '/media_files/'
+FLIERS_PATH = 'fliers'
+STATIC_ROOT = os.path.join(BASE_DIR, 'static/')
+MEDIA_ROOT = os.path.join(BASE_DIR, 'media_files')
```
```python
os.path.join(BASE_DIR, MEDIA_URL)
```
IDK if this would be a good way of putting this but maybe for the variable sake

kevinschoonover commented on ACM_General/core/templates/core/base.html:
```diff
@@ -0,0 +1,85 @@
+{% load static %}
+
+{% block html_tag %}<html lang='en'>{% endblock %}
+    <head>
```
I believe this is using 4 spaces. Clay is trying to use 2 spaces for HTML so for style consistency can we change

kevinschoonover commented on ACM_General/events/models.py:
```diff
@@ -3,92 +3,193 @@
+    :param filename: The filename of the image being used as a flier for the current event being created
+
+    :rtype: string
+    :returns: String that contains a path to where to save the current event flier.
```
String that contains the generated path to save or collect the flier image.

kevinschoonover commented on ACM_General/events/models.py:
```diff
@@ -3,92 +3,193 @@
-# Create your models here.
+def get_path_for_flier(instance, filename):
+    """
+    Used to obtain a path that uses an instance of the Event object to get the "date_hosted" variable.
```
Used to obtain a path based on an instance of the Event (try to make this a cross reference if you can) object pasted into the function.

kevinschoonover commented on ACM_General/events/models.py:
```diff
@@ -3,92 +3,193 @@
-                            auto_now_add=True,
-                            editable=False,
-                   )
+            verbose_name=_('Date Created'),
```
Spaces

kevinschoonover commented on ACM_General/events/models.py:
```diff
@@ -3,92 +3,193 @@
+            related_name='User',
+        )
+
+    #: The SIG hosting the event; represented as a ForeignKey og the SIG model.
```
of*

kevinschoonover commented on ACM_General/events/models.py:
```diff
@@ -3,92 +3,193 @@
+            max_length=256,
+        )
+
+    #: A description of what the event will consist of; as represented as a CharField.
```
"represented as" to keep style consistent

kevinschoonover commented on ACM_General/events/templates/events/create-event.html:
```diff
@@ -0,0 +1,94 @@
+{% endblock %}
+
+{% block body_header %}
+
```
Remove this space

kevinschoonover commented on ACM_General/events/templates/events/create-event.html:
```diff
@@ -0,0 +1,94 @@
+         on the page but that is because I have the items arranged in "row-reverse" order through using FlexBox
+         (tool used for arrangning items on page amongst other things; in the css file).
+    -->
+    <div class="header">
```
2 spaces instead of 4

kevinschoonover commented on ACM_General/events/templates/events/form-template.html:
```diff
@@ -0,0 +1,13 @@
+{% for field in form %}
+    <div class="flex-item">
```
2 spaces instead of 4

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -14,30 +21,30 @@ def setUp(self):
-                            chair=self.user,
-                            description='test',
-                        )
+            founder=self.user,
```
spaces

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -6,6 +6,13 @@
+from django.core.files.uploadedfile import SimpleUploadedFile
+import os
+import shutil
+from datetime import datetime
```
Import order

kevinschoonover commented on ACM_General/events/views.py:
```diff
@@ -1,10 +1,66 @@
-        'events/listEvents.html',
-        {'eventsList': models.Event.objects.all()}
-    ))
+from django.shortcuts import render
```
import order

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -14,30 +21,30 @@ def setUp(self):
+            description='test',
+        )
 
     def test_create_event(self):
```
test docstrings for context please

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -82,39 +90,251 @@ def setUp(self):
+        # that's okay since we're testing to make sure the 'is_active' method works correctly.
+        self.assertFalse(event.is_active)
+
+        event2=models.Event.objects.create(
```
spaces

kevinschoonover commented on ACM_General/events/urls.py:
```diff
@@ -1,22 +1,11 @@
-urlpatterns = [
-    url('^$', views.list_events, name='events-list'),
-]
+from django.conf.urls import url, include
```
space between different import groups

kevinschoonover commented on ACM_General/events/views.py:
```diff
@@ -1,10 +1,66 @@
+
+def create_event(request):
+    """
+    This function is used for authenticating users who have permission to
```
"This view allows authenticated users to create events"

kevinschoonover commented on ACM_General/events/views.py:
```diff
@@ -1,10 +1,66 @@
+              explains the errors that occured when they tried to submit their event.
+    """
+
+    # Used for testing purposes
```
Not needed for develop

kevinschoonover commented on ACM_General/home/static/home/scss/main.scss:
```diff
@@ -241,7 +226,7 @@ nav .nav-link-wrapper {
-  
+
   display: inline-block;
   width: 16.6666666666666666666666666666666666666666666666666666%
```
This isn't a comment for you, but WOAAHHHHHH thats a lot of 6s

kevinschoonover commented on ACM_General/home/templates/home/index.html:
```diff
@@ -2,40 +2,12 @@
 
 {% block head_stylesheets %}
-  <link rel="stylesheet" href="{% static "/home/css/index.css" %}"/>
+	<link rel="stylesheet" href="{% static "home/css/index.css" %}" />
```
2 spaces

kevinschoonover commented on ACM_General/home/urls.py:
```diff
@@ -1,19 +1,6 @@
-"""
 from django.conf.urls import url
+from django.conf import settings
+from django.conf.urls.static import static
```
import spacing

kevinschoonover commented on ACM_General/home/views.py:
```diff
@@ -1,16 +1,23 @@
-    )
+
+    # Grabs the 3 nearest upcoming events that haven't expired yet
+    events = Event.objects.all().order_by('date_hosted').filter(date_expire__gte=timezone.now())
```
Is there any way to do this optimally in the events so it doesn't query all events and then does `len()`? This may prove resource intensive as more events are created. By optimally I mean instead of `all()`.

kevinschoonover commented on ACM_General/rest_api/tests.py:
```diff
@@ -9,11 +9,16 @@
 from sigs.models import SIG
 from payments.models import TransactionCategory, Product, Transaction
 from events.models import Event
+from django.core.files.uploadedfile import SimpleUploadedFile
```
import pep8

kevinschoonover commented on ACM_General/rest_api/tests.py:
```diff
@@ -9,11 +9,16 @@
+from rest_framework.test import APIClient
 
 # Create your tests here.
 class AccountsTestCase(TestCase):
```
Docstrings for test context please

kevinschoonover reviewed (changes requested):
As far as content go, I think its all good. Just a lot of the same style-wise error which I didn't bring up earlier.

Commit:
[FIX] Vagrantfile spacing fix and order of operations fix

- The incorrect spacing in the Vagrantfile on certain lines was
  causing certain lines to not be run.
- I could only see the incorrect spacing by viewing the file while
  inside of vagrant ssh and inside of the vim text editor; didn't
  notice it while on my local computer in the Atom text editor.

Commit:
Merge branch 'createEvent' of github.com:MST-ACM/acm.mst.edu into createEvent

Commit:
[UPDATE] Large update to syntax & other small fixes

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -257,22 +261,22 @@ class EventFormTestCase(TestCase):
         # Sets up image variable for creating Event
         image_path = 'test_data/test_image.jpg'
-        image = SimpleUploadedFile(name='test_image.jpg', content=open(image_path, 'rb').read(), content_type='image/jpeg')
+        image = SimpleUploadedFile(name = 'test_image.jpg', content = open(image_path, 'rb').read(), content_type = 'image/jpeg')
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -257,22 +261,22 @@ class EventFormTestCase(TestCase):
-            'date_hosted': timezone.now()+timezone.timedelta(days=1),
-            'date_expire': timezone.now()+timezone.timedelta(days=2),
+            'date_hosted': timezone.now() + timezone.timedelta(days = 1),
+            'date_expire': timezone.now() + timezone.timedelta(days = 2),
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -317,7 +321,7 @@ def test_event_form_with_no_data(self):
         test_data = self.data
         test_data['date_hosted'] = datetime.min
-        test_data['date_expire'] = datetime.min + timezone.timedelta(days=1)
+        test_data['date_expire'] = datetime.min + timezone.timedelta(days = 1)
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -327,7 +331,7 @@ def test_event_form_with_both_expiration_and_host_date_before_current_time(self)
     def test_event_form_with_expiration_date_before_host_date(self):
         test_data = self.data
-        test_data['date_hosted'] = datetime.min + timezone.timedelta(days=1)
+        test_data['date_hosted'] = datetime.min + timezone.timedelta(days = 1)
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -230,7 +234,7 @@ def test_create_event_function_as_post_request_with_valid_info(self):
     def test_create_event_function_as_post_request_with_invalid_info(self):
         # Logs the superuser in
-        self.client.login(email=self.email)
+        self.client.login(email = self.email)
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -209,14 +213,14 @@ def test_access_create_event_page_with_non_superuser(self):
 
     def test_access_create_event_page_with_superuser(self):
-        self.assertTrue(self.client.login(email=self.email))
+        self.assertTrue(self.client.login(email = self.email))
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -209,14 +213,14 @@ def test_access_create_event_page_with_non_superuser(self):
     def test_create_event_function_as_post_request_with_valid_info(self):
         # Logs the superuser in
-        self.client.login(email=self.email)
+        self.client.login(email = self.email)
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -167,21 +171,21 @@ def setUp(self):
-            description='test',
+        self.user = User.objects.create_superuser(self.email)
+        self.sig = SIG.objects.create_sig(
+            founder = self.user,
```
keyword arguments have no spaces in equals. More PEP8 rules

kevinschoonover commented on ACM_General/events/tests.py:
```diff
@@ -167,21 +171,21 @@ def setUp(self):
         # Sets up image variable for creating Event
         image_path = 'test_data/test_image.jpg'
-        image = SimpleUploadedFile(name='test_image.jpg', content=open(image_path, 'rb').read(), content_type='image/jpeg')
+        image = SimpleUploadedFile(name = 'test_image.jpg', content = open(image_path, 'rb').read(), content_type = 'image/jpeg')
```
keyword arguments have no spaces in equals. More PEP8 rules

Commit:
Fixed keyword spacing in the Event's model and test file

Commit:
[UPDATE] Added new test for homepage and fixed index view function (in home app)

Commit:
Merge branch 'develop' into createEvent
